File: new_kb/parser_samples/sample.c
#include <stdio.h>

typedef struct {
    int x;
    int y;
} Point;

/* add two ints */
int add(int a, int b) {
    return a + b;
}

int main(void) {
    Point p = {1, 2};
    printf("%d\n", add(p.x, p.y));
    return 0;
}

File: new_kb/parser_samples/sample.go
package sample

import (
	"fmt"
)

type Point struct {
	X int
	Y int
}

// Add returns the sum of two points.
func (p Point) Add(other Point) Point {
	return Point{X: p.X + other.X, Y: p.Y + other.Y}
}

func main() {
	p1 := Point{X: 1, Y: 2}
	p2 := Point{X: 3, Y: 4}
	fmt.Println(p1.Add(p2))
}
